resources/sample_vaults/Tasks-Demo/Manual Testing/Smoke Testing the Tasks Plugin - Debug View.md

```diff
@@ -20,3 +20,11 @@ path includes Smoke Testing the Tasks Plugin
 # group by heading
 # short view
 ```
+
+Task SQL Query Version
+
+```tasks-sql
+WHERE path LIKE '%Smoke Testing the Tasks Plugin%'
+# group by heading
+# short view
+```
```

Comment:
It turned out to be really useful to be able to see the pairs of search results next to each other in `Manual Testing/`.

Great idea. Thank you.

This one gives a different appearance from the tasks block, and I'm unsure why.

A couple of comments.... The following was tested one 572a11b08e9347fc79661ec5c177bc0e1a99d284.

---

1. The `tasks-sql` block (green border) displays in short mode, whereas the `tasks` one (blue border) is in full mode.

![image](https://user-images.githubusercontent.com/4840096/232616874-48910d4f-a2b5-4918-9195-6357a833df7d.png)


![image](https://user-images.githubusercontent.com/4840096/232616916-f98c11b3-6078-4b04-b387-df69adedd337.png)

In some other matching pairs I looked at, the display seemed to be the same, so I can't work out why this one is different.

---

2. I thought (wrongly) that the commented-out short-mode instruction might be being parsed... as an experiment I uncommented it out...

~~~

```tasks-sql
WHERE path LIKE '%Smoke Testing the Tasks Plugin%'
# group by heading
short view
```
~~~

And then I got:

![image](https://user-images.githubusercontent.com/4840096/232617592-3368e1b7-6368-417b-941a-478ae4353010.png)


This looks to me like a case where an error message of some sort is displayed in the query.

Any time where `0 tasks` is displayed for some reason other than there actually being 0 tasks in the vault matching a valid query is misleading users, I feel....

Reply:
For point 2 I have seen that when navigating between pages and changing view modes but cannot seem to see where it is getting the idea the empty collection is good to show as the error message is not undefined.

Reply:
For point one, the SQL query uses the `#` value to indicate that the line is not a SQL query but a directive or a comment. So #short view (with or without the space) will indicate that the view should be shortened. Im used to that format so did not realize the tasks query had them commented out. :D

Reply:
updated the block to be explicit.


```tasks-sql
WHERE path LIKE '%Smoke Testing the Tasks Plugin%'
#commented out group by heading
#commented out short view
```

Reply:
Also did a quick debug and error is set to a value, I am relying on the inbuilt render and it picking up error and displaying it. Any gotchas you know if in the rendering code?

Reply:
Not that I’m aware of. Sorry.

Reply:
ok, worked it out. The tasks engine pre-parses the filters before the render call. I had the query executing after render call. So added a pre parse to validate query with no tasks during creation and then on render the error will actually be set. 

Validated that it is working correctly now.

Reply:
> ok, worked it out. The tasks engine pre-parses the filters before the render call. I had the query executing after render call. So added a pre parse to validate query with no tasks during creation and then on render the error will actually be set.
> 
> Validated that it is working correctly now.

Brilliant. Thank you.